ydb/src/load_balancer.rs

```diff
@@ -312,4 +706,473 @@ mod test {
         assert!(*map.get(two.to_string().as_str()).unwrap() > 30);
         Ok(())
     }
+
+    #[test]
+    fn split_by_location() -> YdbResult<()> {
```

Comment:
what about extract test to separate file?

it will be better to separate test/production code for search usages and diffs in PRs/